### PR TITLE
[iOS] Add support for Siri shortcuts

### PR DESCRIPTION
**Pull request by saghul**

### CODE REVIEW

File: ios/app/src/ViewController.m [hunk 1]
```diff
@@ -78,6 +87,36 @@ - (void)conferenceFailed:(NSDictionary *)data {
 
 - (void)conferenceJoined:(NSDictionary *)data {
     [self _onJitsiMeetViewDelegateEvent:@"CONFERENCE_JOINED" withData:data];
+
+    // Register a NSUserActivity for this conference so it can be invoked as a
+    // Siri shortcut. This is only supported in iOS >= 12.
+#ifdef __IPHONE_12_0
+    if (@available(iOS 12.0, *)) {
```
Comment (saghul) [inline ios/app/src/ViewController.m]: The avid reader may wonder why these 2 similar checks are needed. The former (the ifdef) is needed at build time, to make sure we can build the app with Xcode versions without the iOS 12 SDK, and the latter (the if available) is needed at runtime, so devices not running older iOS versions can run the app but not that part of the code.

File: ios/app/src/AppDelegate.m [hunk 2]
```diff
@@ -15,25 +15,50 @@
...
 }
 
 #pragma mark Linking delegate methods
 
--    (BOOL)application:(UIApplication *)application
-  continueUserActivity:(NSUserActivity *)userActivity
-    restorationHandler:(void (^)(NSArray *restorableObjects))restorationHandler {
+- (BOOL) application:(UIApplication *)application
+continueUserActivity:(NSUserActivity *)userActivity
+  restorationHandler:(void (^)(NSArray<id<UIUserActivityRestoring>> * _Nullable))restorationHandler {
```
Comment (saghul) [inline ios/app/src/AppDelegate.m]: This syntax change is suggested by Xcode 10.

**Comment by saghul** [comment]
@lyubomir this is now ready for review.

**Comment by saghul** [comment]
@jitsi/mobile can I get someone to review this? pretty please?

**Comment by paweldomas** [comment]
jenkins test this please

**Comment by paweldomas** [comment]
jenkins test this please

**Comment by paweldomas** [comment]
@saghul I've updated the provisioning profile to include the Siri entitlement, but now there's some permission issue with Travis (not related to this PR).

**Comment by saghul** [comment]
Thanks for the heads up!

**Comment by paweldomas** [comment]
ok the Travis build is working for Siri:
https://travis-ci.org/jitsi/jitsi-meet/builds/445721307

now I need to read and merge the PR

**Comment by saghul** [comment]
@paweldomas Can you please put this back on your radar?

File: ios/app/src/ViewController.m [hunk 3]
```diff
@@ -78,6 +87,36 @@ - (void)conferenceFailed:(NSDictionary *)data {
...
+      NSUserActivity *userActivity
+        = [[NSUserActivity alloc] initWithActivityType:JitsiMeetConferenceActivityType];
+
+      NSString *urlStr = data[@"url"];
+      NSURL *url = [NSURL URLWithString:urlStr];
+      NSString *conference = [url.pathComponents lastObject];
+
+      userActivity.title = [NSString stringWithFormat:@"Join %@", conference];
+      userActivity.suggestedInvocationPhrase = @"Join my awesome meeting";
```
[the remark this answers is not in the document]
Comment (saghul) [reply ios/app/src/ViewController.m]: Sure thing, what do you suggest?

[the remark this answers is not in the document]
Comment (zbettenbuk) [reply ios/app/src/ViewController.m]: Something like "Join my daily status meeting" or similar.

[the remark this answers is not in the document]
Comment (saghul) [reply ios/app/src/ViewController.m]: The meeting can be about anything, not necessarily a "daily status" we shouldn't refer to a specific use case.

**Comment by saghul** [comment]
@zbettenbuk Rebased, PTAL.

File: ios/app/src/ViewController.m (hunk 3 above)
[the remark this answers is not in the document]
Comment (saghul) [reply ios/app/src/ViewController.m]: Note that this is part of the app and not the SDK, so this will only be available to "Jitsi Meet" app users.

**Pull request merged**